arch/arm/include/cortex_m/stack.h

```diff
@@ -48,6 +48,13 @@ static ALWAYS_INLINE void _InterruptStackSetup(void)
 #endif
 
 	__set_MSP(msp);
+#if defined(CONFIG_BUILTIN_STACK_GUARD)
+#if defined(CONFIG_CPU_CORTEX_M_HAS_SPLIM)
+	__set_MSPLIM((u32_t)_interrupt_stack);
+#else
+#error "Built-in MSP limit checks not supported by HW"
```

Comment:
I think the else error checking and double #if defined() is overkill, it seems highly unlikely there will be some new different way to do this on ARM Cortex-M and if so, we will deal with it in the far far future when it might exist.  Or is there some case you can think of already that this is a concern?

Reply:
Thanks  for your comment. The reason for the internal #if is the fact that a Cortex-M23 may _only_ have the Secure versions of the PSPLIM and MSPLIM. For that, we will need to have another option definition: CPU_CORTEX_M_HAS_SPLIM_S. This means that for Cortex-M23 we will only be able to use the SP-based guards if we are building a Secure firmware. Therefore, we will need to have something like:

#if defined(CPU_HAS_SPLIM) || (defined(ARM_SECURE_FIRMWARE) && defined(CPU_HAS_SPLIM_S))

[This will be part of an upcoming PR]